File: HeatLossCalculator.php
<?php

class HeatLossCalculator {
    /**
     * Calculate heat loss.
     *
     * @param float $uValue The U-value (thermal transmittance) of the material (W/m²K).
     * @param float $area The surface area (m²).
     * @param float $tempInside The inside temperature (°C).
     * @param float $tempOutside The outside temperature (°C).
     * @return float The calculated heat loss in Watts (W).
     */
    public function calculate(float $uValue, float $area, float $tempInside, float $tempOutside): float {
        $deltaT = $tempInside - $tempOutside;
        return $uValue * $area * $deltaT;
    }
}

File: heat_loss_calculator/lib/heat_loss_calculator.dart
class HeatLossCalculator {
  /// Calculate heat loss.
  ///
  /// [uValue] The U-value (thermal transmittance) of the material (W/m²K).
  /// [area] The surface area (m²).
  /// [tempInside] The inside temperature (°C).
  /// [tempOutside] The outside temperature (°C).
  /// Returns the calculated heat loss in Watts (W).
  double calculate(double uValue, double area, double tempInside, double tempOutside) {
    double deltaT = tempInside - tempOutside;
    return uValue * area * deltaT;
  }
}
